app/src/main/java/com/asfoundation/wallet/recover/success/RecoveryWalletSuccessBottomSheetFragment.kt

```diff
@@ -0,0 +1,78 @@
+package com.asfoundation.wallet.recover.success
+
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.asf.wallet.R
+import com.asf.wallet.databinding.SettingsPromoCodeSuccessBottomSheetLayoutBinding
+import com.appcoins.wallet.core.arch.SideEffect
+import com.appcoins.wallet.core.arch.SingleStateFragment
+import com.appcoins.wallet.core.arch.ViewState
+import com.appcoins.wallet.feature.promocode.data.repository.PromoCode
+import com.asf.wallet.databinding.RecoveryWalletSuccessBottomSheetLayoutBinding
+import com.asfoundation.wallet.promo_code.bottom_sheet.success.PromoCodeSuccessBottomSheetFragment
+import com.asfoundation.wallet.recover.entry.RecoverEntryNavigator
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class RecoveryWalletSuccessBottomSheetFragment : BottomSheetDialogFragment(),
+  SingleStateFragment<ViewState, SideEffect> {
+
+
+  @Inject
+  lateinit var navigator: RecoverEntryNavigator
+
+  private val views by viewBinding(RecoveryWalletSuccessBottomSheetLayoutBinding::bind)
+
+  companion object {
+
+    const val IS_FROM_ONBOARDING = "is_from_onboarding"
+    @JvmStatic
+    fun newInstance(isFromOnboarding: Boolean): RecoveryWalletSuccessBottomSheetFragment {
+      return RecoveryWalletSuccessBottomSheetFragment()
+        .apply {
+          arguments = Bundle().apply {
+            putSerializable(IS_FROM_ONBOARDING, isFromOnboarding)
+          }
+        }
+    }
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater, container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View = RecoveryWalletSuccessBottomSheetLayoutBinding.inflate(inflater).root
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    views.recoveryWalletBottomButton.setOnClickListener {
+      if (requireArguments().getSerializable(IS_FROM_ONBOARDING) as Boolean) {
+        navigator.navigateToNavigationBar()
```

Comment:
when navigating after the success, from the onboarding recover with password, the app is closing. 
To replicate: onboarding -> "already have a wallet" -> recover from a password backup -> "got it" on the success